src/core/annotation.js

```diff
@@ -1182,6 +1188,14 @@ class WidgetAnnotation extends Annotation {
       return null;
     }
     const value = annotationStorage[this.data.id];
+    if (value === null || value === undefined) {
+      // value === null when a choice widget
+      // hasn't been changed.
+      // The annotation hasn't been rendered
+      // so use the appearance
+      return null;
+    }
+
     if (value === "") {
       return "";
     }
```

Comment:
Wait, why is this pre-existing case special?

If it's indeed necessary to return an empty string here, a comment explaining why would seem like a very good idea!